[BUG] Unable to import FiftyOneDataset that was exported (both through web interface and code).

### Describe the problem

I am unable to import samples from an exported FiftyOneDataset. I will include the logs in the last section. I built a dataset and exported it into a FiftyOneDataset. This was done both using the web interface where you press `\` ` and type `Export Samples -> Media and Labels ->  FiftyOne Dataset` as well as through code like below:

### Code to reproduce issue

```
final = dataset.load_saved_view("finalview")
final.export(
    export_dir='/shared/export/',
    dataset_type=fo.types.FiftyOneDataset,
    export_media=True,
    progress=True
)
```

### System information

-   **OS Platform and Distribution** (e.g., Linux Ubuntu 22.04): WSL2 - Ubuntu 22.04
-   **Python version** (`python --version`): 3.11
-   **FiftyOne version** (`fiftyone --version`): 1.3.1
-   **FiftyOne installed from** (pip or source): Using docker image `voxel51/fiftyone:1.3.1`

### Other info/logs

```
2025-03-11 21:09:48 [2025-03-12 04:09:48 +0000] [8] [ERROR] Error in ASGI Framework
2025-03-11 21:09:48 Traceback (most recent call last):
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/responses.py", line 270, in __call__
2025-03-11 21:09:48     await wrap(partial(self.listen_for_disconnect, receive))
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/responses.py", line 266, in wrap
2025-03-11 21:09:48     await func()
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/responses.py", line 234, in listen_for_disconnect
2025-03-11 21:09:48     message = await receive()
2025-03-11 21:09:48               ^^^^^^^^^^^^^^^
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 113, in receive_or_disconnect
2025-03-11 21:09:48     async with anyio.create_task_group() as task_group:
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 771, in __aexit__
2025-03-11 21:09:48     raise exc_val
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 121, in receive_or_disconnect
2025-03-11 21:09:48     message = await wrap(wrapped_receive)
2025-03-11 21:09:48               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 116, in wrap
2025-03-11 21:09:48     result = await func()
2025-03-11 21:09:48              ^^^^^^^^^^^^
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 50, in wrapped_receive
2025-03-11 21:09:48     msg = await self.receive()
2025-03-11 21:09:48           ^^^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48   File "/usr/local/lib/python3.11/asyncio/queues.py", line 158, in get
2025-03-11 21:09:48     await getter
2025-03-11 21:09:48 asyncio.exceptions.CancelledError: Cancelled by cancel scope 7fdf8fb167d0
2025-03-11 21:09:48 
2025-03-11 21:09:48 During handling of the above exception, another exception occurred:
2025-03-11 21:09:48 
2025-03-11 21:09:48   + Exception Group Traceback (most recent call last):
2025-03-11 21:09:48   |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/_utils.py", line 76, in collapse_excgroups
2025-03-11 21:09:48   |     yield
2025-03-11 21:09:48   |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/responses.py", line 263, in __call__
2025-03-11 21:09:48   |     async with anyio.create_task_group() as task_group:
2025-03-11 21:09:48   |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 767, in __aexit__
2025-03-11 21:09:48   |     raise BaseExceptionGroup(
2025-03-11 21:09:48   | ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
2025-03-11 21:09:48   +-+---------------- 1 ----------------
2025-03-11 21:09:48     | Traceback (most recent call last):
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/hypercorn/asyncio/task_group.py", line 27, in _handle
2025-03-11 21:09:48     |     await app(scope, receive, send, sync_spawn, call_soon)
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/hypercorn/app_wrappers.py", line 34, in __call__
2025-03-11 21:09:48     |     await self.app(scope, receive, send)
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/applications.py", line 112, in __call__
2025-03-11 21:09:48     |     await self.middleware_stack(scope, receive, send)
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 187, in __call__
2025-03-11 21:09:48     |     raise exc
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 165, in __call__
2025-03-11 21:09:48     |     await self.app(scope, receive, _send)
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/cors.py", line 93, in __call__
2025-03-11 21:09:48     |     await self.simple_response(scope, receive, send, request_headers=headers)
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/cors.py", line 144, in simple_response
2025-03-11 21:09:48     |     await self.app(scope, receive, send)
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 181, in __call__
2025-03-11 21:09:48     |     raise app_exc
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 140, in coro
2025-03-11 21:09:48     |     await self.app(scope, receive_or_disconnect, send_no_error)
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
2025-03-11 21:09:48     |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
2025-03-11 21:09:48     |     raise exc
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
2025-03-11 21:09:48     |     await app(scope, receive, sender)
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/routing.py", line 714, in __call__
2025-03-11 21:09:48     |     await self.middleware_stack(scope, receive, send)
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/routing.py", line 734, in app
2025-03-11 21:09:48     |     await route.handle(scope, receive, send)
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/routing.py", line 288, in handle
2025-03-11 21:09:48     |     await self.app(scope, receive, send)
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/endpoints.py", line 41, in dispatch
2025-03-11 21:09:48     |     await response(self.scope, self.receive, self.send)
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/responses.py", line 262, in __call__
2025-03-11 21:09:48     |     with collapse_excgroups():
2025-03-11 21:09:48     |   File "/usr/local/lib/python3.11/contextlib.py", line 158, in __exit__
2025-03-11 21:09:48     |     self.gen.throw(typ, value, traceback)
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/_utils.py", line 82, in collapse_excgroups
2025-03-11 21:09:48     |     raise exc
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/responses.py", line 266, in wrap
2025-03-11 21:09:48     |     await func()
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/responses.py", line 246, in stream_response
2025-03-11 21:09:48     |     async for chunk in self.body_iterator:
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/concurrency.py", line 60, in iterate_in_threadpool
2025-03-11 21:09:48     |     yield await anyio.to_thread.run_sync(_next, as_iterator)
2025-03-11 21:09:48     |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/anyio/to_thread.py", line 56, in run_sync
2025-03-11 21:09:48     |     return await get_async_backend().run_sync_in_worker_thread(
2025-03-11 21:09:48     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 2461, in run_sync_in_worker_thread
2025-03-11 21:09:48     |     return await future
2025-03-11 21:09:48     |            ^^^^^^^^^^^^
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 962, in run
2025-03-11 21:09:48     |     result = context.run(func, *args)
2025-03-11 21:09:48     |              ^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/concurrency.py", line 49, in _next
2025-03-11 21:09:48     |     return next(iterator)
2025-03-11 21:09:48     |            ^^^^^^^^^^^^^^
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/operators/server.py", line 114, in create_response_generator
2025-03-11 21:09:48     |     for generated_message in generator:
2025-03-11 21:09:48     |   File "/fiftyone/default/__plugins__/@voxel51/io/__init__.py", line 183, in execute
2025-03-11 21:09:48     |     for update in _import_media_and_labels(ctx):
2025-03-11 21:09:48     |   File "/fiftyone/default/__plugins__/@voxel51/io/__init__.py", line 862, in _import_media_and_labels
2025-03-11 21:09:48     |     ctx.dataset.add_dir(
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/core/dataset.py", line 5326, in add_dir
2025-03-11 21:09:48     |     return self.add_importer(
2025-03-11 21:09:48     |            ^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/core/dataset.py", line 5922, in add_importer
2025-03-11 21:09:48     |     return foud.import_samples(
2025-03-11 21:09:48     |            ^^^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/utils/data/importers.py", line 123, in import_samples
2025-03-11 21:09:48     |     return dataset_importer.import_samples(
2025-03-11 21:09:48     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/utils/data/importers.py", line 1812, in import_samples
2025-03-11 21:09:48     |     return self._import_samples(
2025-03-11 21:09:48     |            ^^^^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/utils/data/importers.py", line 1890, in _import_samples
2025-03-11 21:09:48     |     dataset._merge_doc(new_doc)
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/core/dataset.py", line 3505, in _merge_doc
2025-03-11 21:09:48     |     _merge_dataset_doc(
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/core/dataset.py", line 8903, in _merge_dataset_doc
2025-03-11 21:09:48     |     dataset._sample_doc_cls.merge_field_schema(
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/core/odm/mixins.py", line 312, in merge_field_schema
2025-03-11 21:09:48     |     validate_field_name(
2025-03-11 21:09:48     |   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/core/odm/utils.py", line 146, in validate_field_name
2025-03-11 21:09:48     |     raise ValueError(
2025-03-11 21:09:48     | ValueError: Invalid field name: '_dataset_id'. Field names cannot start with '_'
2025-03-11 21:09:48     +------------------------------------
2025-03-11 21:09:48 
2025-03-11 21:09:48 During handling of the above exception, another exception occurred:
2025-03-11 21:09:48 
2025-03-11 21:09:48 Traceback (most recent call last):
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/hypercorn/asyncio/task_group.py", line 27, in _handle
2025-03-11 21:09:48     await app(scope, receive, send, sync_spawn, call_soon)
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/hypercorn/app_wrappers.py", line 34, in __call__
2025-03-11 21:09:48     await self.app(scope, receive, send)
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/applications.py", line 112, in __call__
2025-03-11 21:09:48     await self.middleware_stack(scope, receive, send)
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 187, in __call__
2025-03-11 21:09:48     raise exc
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 165, in __call__
2025-03-11 21:09:48     await self.app(scope, receive, _send)
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/cors.py", line 93, in __call__
2025-03-11 21:09:48     await self.simple_response(scope, receive, send, request_headers=headers)
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/cors.py", line 144, in simple_response
2025-03-11 21:09:48     await self.app(scope, receive, send)
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 181, in __call__
2025-03-11 21:09:48     raise app_exc
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 140, in coro
2025-03-11 21:09:48     await self.app(scope, receive_or_disconnect, send_no_error)
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
2025-03-11 21:09:48     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
2025-03-11 21:09:48     raise exc
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
2025-03-11 21:09:48     await app(scope, receive, sender)
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/routing.py", line 714, in __call__
2025-03-11 21:09:48     await self.middleware_stack(scope, receive, send)
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/routing.py", line 734, in app
2025-03-11 21:09:48     await route.handle(scope, receive, send)
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/routing.py", line 288, in handle
2025-03-11 21:09:48     await self.app(scope, receive, send)
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/endpoints.py", line 41, in dispatch
2025-03-11 21:09:48     await response(self.scope, self.receive, self.send)
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/responses.py", line 262, in __call__
2025-03-11 21:09:48     with collapse_excgroups():
2025-03-11 21:09:48   File "/usr/local/lib/python3.11/contextlib.py", line 158, in __exit__
2025-03-11 21:09:48     self.gen.throw(typ, value, traceback)
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/_utils.py", line 82, in collapse_excgroups
2025-03-11 21:09:48     raise exc
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/responses.py", line 266, in wrap
2025-03-11 21:09:48     await func()
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/responses.py", line 246, in stream_response
2025-03-11 21:09:48     async for chunk in self.body_iterator:
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/concurrency.py", line 60, in iterate_in_threadpool
2025-03-11 21:09:48     yield await anyio.to_thread.run_sync(_next, as_iterator)
2025-03-11 21:09:48           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/anyio/to_thread.py", line 56, in run_sync
2025-03-11 21:09:48     return await get_async_backend().run_sync_in_worker_thread(
2025-03-11 21:09:48            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 2461, in run_sync_in_worker_thread
2025-03-11 21:09:48     return await future
2025-03-11 21:09:48            ^^^^^^^^^^^^
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 962, in run
2025-03-11 21:09:48     result = context.run(func, *args)
2025-03-11 21:09:48              ^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/starlette/concurrency.py", line 49, in _next
2025-03-11 21:09:48     return next(iterator)
2025-03-11 21:09:48            ^^^^^^^^^^^^^^
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/operators/server.py", line 114, in create_response_generator
2025-03-11 21:09:48     for generated_message in generator:
2025-03-11 21:09:48   File "/fiftyone/default/__plugins__/@voxel51/io/__init__.py", line 183, in execute
2025-03-11 21:09:48     for update in _import_media_and_labels(ctx):
2025-03-11 21:09:48   File "/fiftyone/default/__plugins__/@voxel51/io/__init__.py", line 862, in _import_media_and_labels
2025-03-11 21:09:48     ctx.dataset.add_dir(
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/core/dataset.py", line 5326, in add_dir
2025-03-11 21:09:48     return self.add_importer(
2025-03-11 21:09:48            ^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/core/dataset.py", line 5922, in add_importer
2025-03-11 21:09:48     return foud.import_samples(
2025-03-11 21:09:48            ^^^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/utils/data/importers.py", line 123, in import_samples
2025-03-11 21:09:48     return dataset_importer.import_samples(
2025-03-11 21:09:48            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/utils/data/importers.py", line 1812, in import_samples
2025-03-11 21:09:48     return self._import_samples(
2025-03-11 21:09:48            ^^^^^^^^^^^^^^^^^^^^^
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/utils/data/importers.py", line 1890, in _import_samples
2025-03-11 21:09:48     dataset._merge_doc(new_doc)
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/core/dataset.py", line 3505, in _merge_doc
2025-03-11 21:09:48     _merge_dataset_doc(
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/core/dataset.py", line 8903, in _merge_dataset_doc
2025-03-11 21:09:48     dataset._sample_doc_cls.merge_field_schema(
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/core/odm/mixins.py", line 312, in merge_field_schema
2025-03-11 21:09:48     validate_field_name(
2025-03-11 21:09:48   File "/opt/.fiftyone-venv/lib/python3.11/site-packages/fiftyone/core/odm/utils.py", line 146, in validate_field_name
2025-03-11 21:09:48     raise ValueError(
2025-03-11 21:09:48 ValueError: Invalid field name: '_dataset_id'. Field names cannot start with '_'
```

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another
member of your organization be willing to contribute a fix for this bug to the
FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently
-   [ ] Yes. I would be willing to contribute a fix for this bug with guidance
        from the FiftyOne community
-   [X] No. I cannot contribute a bug fix at this time

--- a/fiftyone/core/odm/mixins.py
+++ b/fiftyone/core/odm/mixins.py
@@ -1130,9 +1130,6 @@ class DatasetMixin(object):
                     )
                 )
 
-        if isinstance(field, fof.ObjectIdField) and field_name.startswith("_"):
-            field_name = field_name[1:]
-
         if field_name in doc._fields:
             existing_field = doc._fields[field_name]
 

--- a/tests/unittests/import_export_tests.py
+++ b/tests/unittests/import_export_tests.py
@@ -3479,6 +3479,35 @@ class MultitaskImageDatasetTests(ImageDatasetTests):
             )
         )
 
+        # Test full schema merge
+
+        export_dir = self._new_dir()
+
+        sample1 = fo.Sample(filepath="video1.mp4")
+        sample1.frames[1] = fo.Frame()
+
+        dataset1 = fo.Dataset()
+        dataset1.add_sample(sample1)
+
+        sample2 = fo.Sample(filepath="video2.mp4")
+        sample2.frames[1] = fo.Frame()
+
+        dataset2 = fo.Dataset()
+        dataset2.add_sample(sample2)
+
+        dataset2.export(
+            export_dir=export_dir,
+            dataset_type=fo.types.FiftyOneDataset,
+            export_media=False,
+        )
+
+        dataset1.add_dir(
+            dataset_dir=export_dir,
+            dataset_type=fo.types.FiftyOneDataset,
+        )
+
+        self.assertEqual(len(dataset1), 2)
+
     @skipwindows
     @drop_datasets
     def test_legacy_fiftyone_dataset(self):
@@ -3842,6 +3871,35 @@ class MultitaskImageDatasetTests(ImageDatasetTests):
                 m1 < m2 for m1, m2 in zip(last_modified_at1, last_modified_at2)
             )
         )
+
+        # Test full schema merge
+
+        export_dir = self._new_dir()
+
+        sample1 = fo.Sample(filepath="video1.mp4")
+        sample1.frames[1] = fo.Frame()
+
+        dataset1 = fo.Dataset()
+        dataset1.add_sample(sample1)
+
+        sample2 = fo.Sample(filepath="video2.mp4")
+        sample2.frames[1] = fo.Frame()
+
+        dataset2 = fo.Dataset()
+        dataset2.add_sample(sample2)
+
+        dataset2.export(
+            export_dir=export_dir,
+            dataset_type=fo.types.LegacyFiftyOneDataset,
+            export_media=False,
+        )
+
+        dataset1.add_dir(
+            dataset_dir=export_dir,
+            dataset_type=fo.types.LegacyFiftyOneDataset,
+        )
+
+        self.assertEqual(len(dataset1), 2)
 
 
 class OpenLABELImageDatasetTests(ImageDatasetTests):

